src/CodeGenerator/src/Generator/OperationGenerator.php

```diff
@@ -198,6 +198,9 @@ private function setMethodBody(Method $method, Operation $operation, ClassName $
             $classBuilder->addUse($errorClass->getFqdn());
 
             $mapping[] = sprintf('%s => %s::class,', var_export($error->getCode() ?? $error->getName(), true), $errorClass->getName());
+            if ($operation->getHttpMethod() === 'HEAD' && ($statusCode = $error->getStatusCode()) >= 400) {
```

Comment:
this should declare a dependency on `async-aws/core` at version `^1.22` inside this `if`, as this will be the version of `async-aws/core` that contains the logic making it work (otherwise, the bug in async-aws/s3 might not be fixed if it installs an older version of the core).

We have the infrastructure to handle that properly:
```suggestion
            if ($operation->getHttpMethod() === 'HEAD' && ($statusCode = $error->getStatusCode()) >= 400) {
                $this->requirementsRegistry->addRequirement('async-aws/core', '^1.22');
```

Reply:
good catch, thanks for the hint